### PR TITLE
nix: enable auto-optimize-store

### PR DESCRIPTION
**Pull request by Mic92**

### CODE REVIEW

File: nixos/common/nix.nix [hunk 1]
```diff
@@ -13,6 +13,9 @@
   # The default at 10 is rarely enough.
   nix.settings.log-lines = lib.mkDefault 25;
 
+  # De-duplicate store paths using hardlinks
+  nix.settings.auto-optimise-store = lib.mkDefault true;
```
Comment (SuperSandro2000) [inline nixos/common/nix.nix:17]: In my experience this still slows down copies of store paths significantly. IMO the systemd timer is a better alternative which doesn't slow down every store operation.

Comment (Mic92) [reply nixos/common/nix.nix:17]: Makes sense. We can have the timer also on both platforms.

Comment (Mic92) [reply nixos/common/nix.nix:17]: Is there an option for this?

Comment (SuperSandro2000) [reply nixos/common/nix.nix:17]: yep https://search.nixos.org/options?channel=unstable&show=nix.optimise.automatic&from=0&size=50&sort=relevance&type=packages&query=nix.optimise

Comment (Mic92) [reply nixos/common/nix.nix]: Applied.

File: nixos/common/nix.nix [hunk 2]
```diff
@@ -13,6 +13,9 @@
   # The default at 10 is rarely enough.
   nix.settings.log-lines = lib.mkDefault 25;
 
+  # De-duplicate store paths using hardlinks
+  nix.optimise.automatic = true;
```
Comment (zowoq) [inline nixos/common/nix.nix:17]:
```suggestion
  nix.optimise.automatic = lib.mkDefault true;
```

**Comment by Mic92** [comment]
@mergify rebase

**Comment by Mic92** [comment]
@mergify queue

**Pull request merged**